src/types.jl

```diff
@@ -41,6 +42,11 @@ Custom neighbor finders should sub-type this type.
 """
 abstract type AbstractNeighborFinder end
 
+"""
+A constraint for bonds and angles.
+"""
+abstract type Constraint end
```

Comment:
Unless it's necessary I think it's a better idea not to introduce a new abstract type. I've been meaning to remove the remaining unnecessary abstract types and will do at some point.